### PR TITLE
Hashtag 7 (Modified stream tag and added faz backend functionality)

### PR DESCRIPTION
**Pull request by wsuppiger**
Hi Fayez,

I realized that my last implementation of `StreamTag` in Caw Functions needed a wrapper callback to put the caw object in a HashtagReply before returning it up to the Faz layer.  This makes it so I can keep a generic streamEvent function at the Faz layer, and also by modifying the callback in the server backend it can now interpret the stream reply in any way it likes.  I realize the callback argument of an `Any` instead of an `EventReply` might not be perfect, so I might change this later if it becomes an issue.  

Thanks!

### CODE REVIEW

**Comment by wsuppiger** [comment]
Also had to add a new map `caw_stream_functions_` since they hold a different function signature, and I had to modify the hook function since it needed to check if the function existed in `caw_stream_functions_`.

**Review by floan — APPROVED** [review]
Sorry for the late review, this looks good to me. In general all your code is making sense and you are doing well, from a design point of view though, I would recommend trying to use callback functions a little less if you can. They make the entire code path a little harder to follow. Consider passing in references that can then be populated?

**Comment by wsuppiger** [comment]
Yeah, I agree that all the callbacks can be hard to follow.  I just talked to Adam, and all of the callbacks are necessary since it is the only way possible without long pulling.
